contrib/kafka/pkg/dispatcher/dispatcher.go

```diff
@@ -179,28 +190,53 @@ func (d *KafkaDispatcher) subscribe(channelRef provisioners.ChannelReference, su
 	}
 	channelMap[sub] = consumer
 
-	go func() {
-		for {
-			msg, more := <-consumer.Messages()
-			if more {
-				d.logger.Info("Dispatching a message for subscription", zap.Any("channelRef", channelRef), zap.Any("subscription", sub))
-				message := fromKafkaMessage(msg)
-				err := d.dispatchMessage(message, sub)
-				if err != nil {
-					d.logger.Warn("Got error trying to dispatch message", zap.Error(err))
+	if cluster.ConsumerModePartitions == d.kafkaCluster.GetConsumerMode() {
```

Comment:
I am wondering if we should break these apart into two functions and then `go c.consumerModeLoop()` or whatever on  the if and then `go c.elseGoodNameLoop()` rather than inline the go funcs